VoiceMemos/Controller/VoicesTableViewController.swift

```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
         }
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
```

Comment:
The method is executed on the main thread, so the `dispatch_async ` call is unnecessary.